scripts/build.sh

```diff
@@ -11,12 +11,14 @@ BASE_ARGS="\
     -DENABLE_JIMPLE_FRONTEND=On \
     -DCMAKE_INSTALL_PREFIX:PATH=$PWD/release \
 "
+# Must disable old frontend to enable goto contractor.
 SOLVER_FLAGS="\
     -DENABLE_BOOLECTOR=On \
     -DENABLE_YICES=Off \
     -DENABLE_CVC4=OFF \
     -DENABLE_BITWUZLA=On \
-    -DENABLE_GOTO_CONTRACTOR=OFF \
+    -DENABLE_GOTO_CONTRACTOR=On \
+    -DACADEMIC_BUILD=On \
```

Comment:
I'm sorry for seeing this just now, but what is the purpose of ACADEMIC_BUILD? All I can find is that if cmake downloads dependencies, then it enables Ibex being built against [soplex](https://soplex.zib.de/). That thing is Apache-2.0-licensed as well as ESBMC. (Why) do we need that option?

Reply:
I think its for licencing purposes but I am not sure of the details.
@rafaelsamenezes can explain it better.

Reply:
As stated in the [their docs](https://ibex-lib.readthedocs.io/en/latest/install.html#linux-and-macos): 

_"Soplex is under [ZIB](http://scip.zib.de/academic.txt) academic licence. If you intend to use Ibex with Soplex commercially, you may consider contacting Soplex for a commercial licence."_

We should probably disable this for now as we need to inform the user somehow.

Reply:
Although... the latest version of soplex is now apache. So we could try to compile manually and just point it during ibex compilation.

Reply:
Do we have any use for an soplex-backed ibex? Does it make any difference? Anyways, I'm fine with keeping it for now, though the user should indeed be informed in case this soplex thing makes any measurable difference for ESBMC. The build.sh script passing this option could write a warning...

Reply:
From [ibex manual 2.8](https://ibex-lib.readthedocs.io/en/latest/install.html#standard-install).
> Warning The current release of Ibex is compatible with Soplex 3.1.1

Not sure why it's a warning. Maybe a typo.

Also:
>Warning Soplex is a tierce library under [ZIB](http://scip.zib.de/academic.txt) licence, not LGPL. By configuring Ibex with Soplex, you implicitly accept the terms of this licence.

>If you don’t accept these terms, you can configure Ibex without Soplex:
>~/Ibex/ibex-2.8.9/$ sudo ./waf configure
>However, the performances will be strongly degraded, especially for the global optimizer.

From [soplex license](https://soplex.zib.de/index.php#license):
>Releases up to and including Version 6.0.2 remain under the [ZIB Academic License](https://soplex.zib.de/academic.txt) as indicated by the files contained in the releases. The new license applies from Version 6.0.3 onwards.

The soplex downloaded automatically is 4.0.2 
I will run some benchmarks to see if there is any difference in performance with and without soplex. 
I will also try using ibex with latest soplex and see if it works. 
For now, I will add the warning in the findIbex cmake file and build.sh.